### PR TITLE
feat(information_schema): implement `table_factory` method

### PR DESCRIPTION
**Pull request by zhongzc**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

* Add `DataSource` trait
* impl `DataSource` for `InformationTable`
* impl `table_factory` method for `InformationTable`

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

https://github.com/GreptimeTeam/greptimedb/issues/2065

### CODE REVIEW

**Comment by zhongzc** [comment]
PTAL @waynexia

**Review by waynexia — APPROVED** [review]
LGTM 👍

File: src/table/src/data_source.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+use store_api::storage::ScanRequest;
+
+use crate::error::Result;
+
+// This trait represents a common data source abstraction which provides an interface
+// for retrieving data in the form of a stream of record batches.
+pub trait DataSource {
+    // Retrieves a stream of record batches based on the provided scan request.
+    fn get_stream(&self, request: ScanRequest) -> Result<SendableRecordBatchStream>;
+}
```
Comment (waynexia) [inline src/table/src/data_source.rs:27]:
```suggestion
/// This trait represents a common data source abstraction which provides an interface
/// for retrieving data in the form of a stream of record batches.
pub trait DataSource {
    /// Retrieves a stream of record batches based on the provided scan request.
    fn get_stream(&self, request: ScanRequest) -> Result<SendableRecordBatchStream>;
}
```

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**